src/databricks/labs/ucx/source_code/notebooks/sources.py

```diff
@@ -367,29 +369,22 @@ def __init__(
         path_lookup: PathLookup,
         session_state: CurrentSessionState,
         path: Path,
+        inherited_tree: Tree | None = None,
         content: str | None = None,
     ):
         self._ctx: LinterContext = ctx
         self._path_lookup = path_lookup
         self._session_state = session_state
-        self._path: Path = path
+        self._path = path
+        self._inherited_tree = inherited_tree
         self._content = content
 
     @cached_property
     def _source_code(self) -> str:
         if self._content is None:
-            self._content = self._path.read_text(_guess_encoding(self._path))
+            self._content = self._path.read_text(guess_encoding(self._path))
         return self._content
 
-    def _file_language(self):
```

Comment:
moved to base.by to avoid cyclic imports between cells.py, loaders.py, sources.py and files.py